### PR TITLE
Small fix: optional DeepRC/TCRDist dependencies

### PR DESCRIPTION
**Pull request by BlueBasilisk**
Addresses this trello card: https://trello.com/c/gqK6DsfW/312-look-into-whether-it-could-be-easy-to-separate-tcrdist-and-deeprc-from-the-core-installation-procedure-to-have-them-as-add-ons-m

- DeepRC and TCRDist are now by default not installed, but can be added. 
- Installation documentation has been updated 
- Installation docs is also now written to a version with and without R, to make it clearer
- Even more proof that this separation of dependencies is necessary: I encountered a bug when installing the TCRDist dependencies on Mac. In other words, installing the current version of immuneML did not work without errors! But now that TCRDist is optional, that bug only occurs when explicitly adding TCRDist. I tried to find a fix for this (automatically installing other packages), but did not manage to find a good solution. In a way, it is not our responsibility. In the FAQ I added a link to the github issue where someone proposes a hacky workaround, in case they are super dedicated to make it work...

### CODE REVIEW

**Comment by BlueBasilisk** [comment]
After discussing with Milena, it turns out parasail 1.2 does work. I will update this.

**Comment by BlueBasilisk** [comment]
adding parasail==1.2 explicitly in setup.py resolved the issue with TCRDist! This has been updated now, also in the docs.

File: docs/source/installation/install_with_package_manager.rst [hunk 1]
```diff
@@ -1,7 +1,79 @@
...
+
+  pip install git+https://github.com/uio-bmi/immuneML
+
+Alternatively, if you want to install immuneML including optional extras (:code:`DeepRC`, :code:`TCRDist`), use:
+
+.. code-block:: console
+
+  pip install git+https://github.com/uio-bmi/immuneML#egg=immuneML[DeepRC,TCRDist]
+
+Installing DeepRC and TCRDist dependencies is necessary to use the :ref:`DeepRC` and :ref:`TCRDISTClassifier` ML methods.
```
Comment (pavlovicmilena) [inline docs/source/installation/install_with_package_manager.rst:44]: it is also necessary for TCRdistMotifDiscovery report.

File: docs/source/installation/install_with_package_manager.rst [hunk 2]
```diff
@@ -10,36 +82,37 @@ Steps to install immuneML with Anaconda (tested with version 4.8.3):
...
 
-Alternatively, if you want to install immuneML including all R plots, use:
+Alternatively, if you want to install immuneML including optional extras (:code:`DeepRC`, :code:`TCRDist`), use:
 
 .. code-block:: console
 
-  pip install git+https://github.com/uio-bmi/immuneML#egg=immuneML[R_plots]
+  pip install git+https://github.com/uio-bmi/immuneML#egg=immuneML[R_plots,DeepRC,TCRDist]
+
+Installing DeepRC and TCRDist dependencies is necessary to use the :ref:`DeepRC` and :ref:`TCRDISTClassifier` ML methods.
```
Comment (pavlovicmilena) [inline docs/source/installation/install_with_package_manager.rst:115]: the same as above :)

File: docs/source/installation/install_with_package_manager.rst (hunk 1 above)
Comment (BlueBasilisk) [reply docs/source/installation/install_with_package_manager.rst]: updated :)

**Review by pavlovicmilena — APPROVED** [review]

**Pull request merged**